components/gcp-click-to-deploy/src/DeployForm.tsx

```diff
@@ -121,6 +123,13 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
     // be able to click submit until the fetches have succeeded. How can we do
     // that?
 
+    const params = new URLSearchParams(this.props.location.search);
+    if (params.get('version')) {
+      this._query_string_version = params.get('version');
```

Comment:
Why the extra variable? Why not just append/prepend to the versions array? Then the rest of the render logic can remain as is: render all items in the list, and auto-select the first/last one.

Reply:
kunming mentioned the REACT_APP_VERSIONS will take precedence. The code might get ugly without this extra variable.

Reply:
Suggestion: This logic should just be done at the beginning of the render function rather than here, it'd be cleaner to first compute the list then render the entire array here.

Reply:
done